src/index.js

```diff
@@ -415,6 +415,70 @@ walletRouter.post('/simple-send-tx',
   }
 });
 
+
+/**
+ * Method that select utxos to be used in a transaction
+ * following the filter query sent in the request
+ *
+ * XXX this method should be created in the lib itself
+ * however the current version of the lib is already incompatible with
+ * the method calls we have right now after some refactors were done
+ * to use the facada in the wallets mobile/desktop.
+ * Because of that I decided to create this method here for now,
+ * not only because it would require more changes to fix the incompatibilities but
+ * we will change even more in the following weeks when we finish the integration
+ * of the lib with the wallet service and the signature methods alignment.
+ * Then part of the work here would be lost anyway in the following weeks.
+ *
+ * I leave this comment here because, as soon as we finish the wallet service integration in the lib
+ * and align the methods signatures, we will fix the incompatibilities in the headless and then we
+ * can add this method in the lib
+ *
+ * @param {HathorWallet} wallet The wallet object
+ * @param {number} sumOutputs The sum of outputs of the transaction I need to fill
+ * @param {Object} options The options to filter the utxos (see utxo-filter API to see the possibilities)
+ */
+const getUtxosToFillTx = (wallet, sumOutputs, options) => {
+  const utxosDetails = wallet.getUtxos(options);
+  // If I can't fill all the amount with the returned utxos, then return null
+  if (utxosDetails.total_amount_available < sumOutputs) {
+    return null;
+  }
+
+  const utxos = utxosDetails.utxos;
+  // Sort utxos with larger amounts first
+  utxos.sort((a, b) => b.amount - a.amount);
+
+  if (utxos[0].amount > sumOutputs) {
```

Comment:
I feel we already have something similar when selecting utxos without queries. Can't we reuse that piece of code?

Reply:
We have a method that does something similar in the lib. The method is `getInputsFromAmount` from `src/wallet.js`.

However this method can't be used as we want and, as I described in the PR, the best approach would be to create a new method in the lib for this but the current version of the lib is not compatible with the headless because of other refactors. The work to make it compatible is not small and might be lost (because the refactor in the facades is not over yet). That's why I decided to create this new method in the headless itself.

Reply:
Ok. So this code duplication will be fixed when we migrate this part to the lib, right? Have you opened an issue to get this migration done later?

Reply:
Yes.